mungegithub/mungers/sig-mention-handler.go

```diff
@@ -0,0 +1,104 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"github.com/golang/glog"
+	//githubapi "github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+	"k8s.io/test-infra/mungegithub/features"
+	github "k8s.io/test-infra/mungegithub/github"
+	//"reflect"
+	"strings"
+)
+
+var (
+	closed          = "closed"
+	currentVersion  = "v1.7"
+	noSigMentionMsg = "There are no sig labels on this issue. Please add a sig label by mentioning a sig[https://github.com/kubernetes/test-infra/blob/master/commands.md]"
+	sigPrefix       = "sig/"
+	needsSigLabel   = "needs-sig"
+)
+
+type SigMentionHandler struct{}
+
+func init() {
+	h := &SigMentionHandler{}
+	RegisterMungerOrDie(h)
+}
+
+// Name is the name usable in --pr-mungers
+func (*SigMentionHandler) Name() string { return "sig-mention-handler" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (*SigMentionHandler) RequiredFeatures() []string {
+	return []string{}
+}
+
+// Initialize will initialize the munger
+func (s *SigMentionHandler) Initialize(config *github.Config, features *features.Features) error {
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (*SigMentionHandler) EachLoop() error { return nil }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (*SigMentionHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+// The algorithm:
+// (1) return false if it is not an issue
+// (2) return false if the issue is closed
+// (3) return false if the issue does not have a v1.7 milestone
+// (4) return false if the issue has any label starting with "sig/"
+// (5) return true otherwise
+func (*SigMentionHandler) IsSigLabelNeeded(obj *github.MungeObject) bool {
+	if obj.Issue == nil || obj.IsPR() || obj.Issue.State == nil || *obj.Issue.State == closed {
+		return false
+	}
+
+	labels := obj.Issue.Labels
+
+	for i := range labels {
+		if labels[i].Name != nil && strings.HasPrefix(*labels[i].Name, sigPrefix) {
+			return false
+		}
+	}
+
+	return true
+}
+
+// Munge is the workhorse notifying issue owner to add a @kubernetes/sig mention if there is none
+func (s *SigMentionHandler) Munge(obj *github.MungeObject) {
+	if !s.IsSigLabelNeeded(obj) {
+		return
+	}
+
+	obj.AddLabel(needsSigLabel)
+
+	msg := ""
+
+	if obj.Issue.User.Login != nil && *obj.Issue.User.Login != "" {
+		msg += "@" + *obj.Issue.User.Login + " "
+	}
+
+	msg += noSigMentionMsg
+
+	if err := obj.WriteComment(msg); err != nil {
+		glog.Errorf("failed to write comment to %v: %q: %v", *obj.Issue.Number, msg, err)
```

Comment:
Rather than reprinting the whole msg, consider making the error message more generic, for example:
`Errorf("failed to leave comment for %s that %v a needs sig label", login, number)`